### PR TITLE
Added Module MapPreview see issue #143

### LINKED ISSUE #143: Map preview
when you hover over a map make a box thing show up that shows the contents of the map

### PR DESCRIPTION
**Pull request by TacticalFacePlant**
This pull is regarding Issue#143

Closes #143

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import me.zeroeightsix.kami.setting.Settings
+import net.minecraft.client.renderer.GlStateManager
+import net.minecraft.item.ItemMap
+import net.minecraft.world.World
+import net.minecraftforge.client.event.RenderTooltipEvent
+
+
+/**
+ * Created by fred41 on 27/05/2020.
+ *
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:16]: could you please remove this extra line?

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+class MapPeview : Module() {
+    private val scale = register(Settings.doubleBuilder("scale").withRange(0.0, 3.0).withValue(0.5).build())
+
+    @EventHandler
+    private val tooltipEventListener = Listener(EventHook { event: RenderTooltipEvent.PostText ->
+        if (event.getStack().getItem() is ItemMap) {
+            val mapdata = (event.getStack().getItem() as ItemMap).getMapData(event.getStack(), mc.world as World)
+            if(mapdata==null)
+                return@EventHook
+            GlStateManager.disableDepth()
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:32]: a bit nit picky, but change it to this please
```kotlin
if (mapdata == null) return@EventHook
```
and leave a space before the GL stuff

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Created by fred41 on 27/05/2020.
+ *
+ */
+@Module.Info(
+        name = "MapPreview",
+        category = Module.Category.RENDER,
+        description = "Shows the contents of a map when you hover over one"
+)
+class MapPeview : Module() {
+    private val scale = register(Settings.doubleBuilder("scale").withRange(0.0, 3.0).withValue(0.5).build())
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:24]: your value can't be bigger then your range, did you mean to make the range 0.0 to 0.5?

**Comment by dominikaaaa** [comment]
Looks good, just a few minor formatting changes.

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt (hunk 3 above)
Comment (FriedrichM) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:24]: it thought this ment with a range form 0 to 3  with default value 0.5

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt (hunk 2 above)
Comment (FriedrichM) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:32]: i can run the default intellij reformat code  is that ok ?

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt (hunk 3 above)
Comment (dominikaaaa) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:24]: oh sorry, I misread that.

File: src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt (hunk 2 above)
Comment (dominikaaaa) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/MapPeview.kt:32]: sure.

**Pull request merged**